#### Importing necessary libraries

In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

#### Importng dataset

In [2]:
df=pd.read_csv('./athlete_events.csv')
df.head(5)


ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

#### Droping unwanted columns

In [3]:
df.drop(columns=['ID','Games'],inplace=True)

#### Studying dataset

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 13 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   Name    271116 non-null  object 
 1   Sex     271116 non-null  object 
 2   Age     261642 non-null  float64
 3   Height  210945 non-null  float64
 4   Weight  208241 non-null  float64
 5   Team    271116 non-null  object 
 6   NOC     271116 non-null  object 
 7   Year    271116 non-null  int64  
 8   Season  271116 non-null  object 
 9   City    271116 non-null  object 
 10  Sport   271116 non-null  object 
 11  Event   271116 non-null  object 
 12  Medal   39783 non-null   object 
dtypes: float64(3), int64(1), object(9)
memory usage: 26.9+ MB


In [5]:
df.describe()

Age         Height         Weight           Year
count  261642.000000  210945.000000  208241.000000  271116.000000
mean       25.556898     175.338970      70.702393    1978.378480
std         6.393561      10.518462      14.348020      29.877632
min        10.000000     127.000000      25.000000    1896.000000
25%        21.000000     168.000000      60.000000    1960.000000
50%        24.000000     175.000000      70.000000    1988.000000
75%        28.000000     183.000000      79.000000    2002.000000
max        97.000000     226.000000     214.000000    2016.000000

In [6]:
df.size

3524508

In [7]:
df.shape

(271116, 13)

In [8]:
df.columns

Index(['Name', 'Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Year',
       'Season', 'City', 'Sport', 'Event', 'Medal'],
      dtype='object')

#### Checking for duplicates

In [9]:
df.duplicated().sum()

1385

### Finding insights from the data 

#### feature engineering using the column medal

In [10]:
df.Medal.unique()

array([nan, 'Gold', 'Bronze', 'Silver'], dtype=object)

In [11]:
##### Lets create seperate column for each medal
df['Gold']=[1 if medal=='Gold' else 0 for medal in df['Medal']]
df['Silver']=[1 if medal=='Silver' else 0 for medal in df['Medal']]
df['Bronze']=[1 if medal=='Bronze' else 0 for medal in df['Medal']]

In [12]:
df.head(5)

Name Sex   Age  Height  Weight            Team  NOC  \
0                 A Dijiang   M  24.0   180.0    80.0           China  CHN   
1                  A Lamusi   M  23.0   170.0    60.0           China  CHN   
2       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark  DEN   
3      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden  DEN   
4  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands  NED   

   Year  Season       City          Sport                             Event  \
0  1992  Summer  Barcelona     Basketball       Basketball Men's Basketball   
1  2012  Summer     London           Judo      Judo Men's Extra-Lightweight   
2  1920  Summer  Antwerpen       Football           Football Men's Football   
3  1900  Summer      Paris     Tug-Of-War       Tug-Of-War Men's Tug-Of-War   
4  1988  Winter    Calgary  Speed Skating  Speed Skating Women's 500 metres   

  Medal  Gold  Silver  Bronze  
0   NaN     0       0       0  
1   NaN     0       0       0  
2   NaN     0       0       0  
3  Gold     1       0       0  
4   NaN     0       0       0

### 1.Which country won most medal

In [13]:
country_wise_medal=df.groupby(['Team'])['Gold', 'Silver','Bronze'].sum()

In [14]:
country_wise_medal

Gold  Silver  Bronze
Team                                       
30. Februar               0       0       0
A North American Team     0       0       4
Acipactli                 0       0       0
Acturus                   0       0       0
Afghanistan               0       0       2
...                     ...     ...     ...
Zambia                    0       1       1
Zefyros                   0       0       0
Zimbabwe                 17       4       1
Zut                       0       3       0
rn-2                      0       0       0

[1184 rows x 3 columns]

In [15]:
#Adding a column  Total to the object country_wise_medal

In [16]:
country_wise_medal['Total']=country_wise_medal['Gold']+country_wise_medal['Silver']+country_wise_medal['Bronze']


In [17]:
country_wise_medal

Gold  Silver  Bronze  Total
Team                                              
30. Februar               0       0       0      0
A North American Team     0       0       4      4
Acipactli                 0       0       0      0
Acturus                   0       0       0      0
Afghanistan               0       0       2      2
...                     ...     ...     ...    ...
Zambia                    0       1       1      2
Zefyros                   0       0       0      0
Zimbabwe                 17       4       1     22
Zut                       0       3       0      3
rn-2                      0       0       0      0

[1184 rows x 4 columns]

In [18]:
country_wise_medal.sort_values(by='Total',ascending=False)

Gold  Silver  Bronze  Total
Team                                      
United States  2474    1512    1233   5219
Soviet Union   1058     716     677   2451
Germany         679     627     678   1984
Great Britain   519     582     572   1673
France          455     518     577   1550
...             ...     ...     ...    ...
Ireland-1         0       0       0      0
Israel-1          0       0       0      0
Israel-2          0       0       0      0
Italy-3           0       0       0      0
rn-2              0       0       0      0

[1184 rows x 4 columns]

In [19]:
country_wise_medal.nlargest(5,'Total')

Gold  Silver  Bronze  Total
Team                                      
United States  2474    1512    1233   5219
Soviet Union   1058     716     677   2451
Germany         679     627     678   1984
Great Britain   519     582     572   1673
France          455     518     577   1550

#### :America got most no of medal

### 2.Country with no gold medal

In [20]:
no_gold_medal=country_wise_medal[country_wise_medal["Gold"]==0]
no_gold_medal


Gold  Silver  Bronze  Total
Team                                              
30. Februar               0       0       0      0
A North American Team     0       0       4      4
Acipactli                 0       0       0      0
Acturus                   0       0       0      0
Afghanistan               0       0       2      2
...                     ...     ...     ...    ...
Yugoslavia-2              0       0       0      0
Zambia                    0       1       1      2
Zefyros                   0       0       0      0
Zut                       0       3       0      3
rn-2                      0       0       0      0

[942 rows x 4 columns]

#### :There are 942 countries with no gold medal

### 3.Is India included in the above list

In [22]:
India_medal=df.query("(Gold==1 or Silver==1 or Bronze==1 )& NOC=='IND'")
India_medal

Name Sex   Age  Height  Weight   Team  NOC  Year  \
4732               Shaukat Ali   M  30.0     NaN     NaN  India  IND  1928   
4736          Syed Mushtaq Ali   M  22.0   165.0    61.0  India  IND  1964   
5032       Richard James Allen   M  25.0   172.0     NaN  India  IND  1928   
5033       Richard James Allen   M  30.0   172.0     NaN  India  IND  1932   
5034       Richard James Allen   M  34.0   172.0     NaN  India  IND  1936   
...                        ...  ..   ...     ...     ...    ...  ...   ...   
233749      Dung Dung Sylvanus   M  31.0   160.0    62.0  India  IND  1980   
236445  Carlyle Carrol Tapsell   M  23.0   182.0     NaN  India  IND  1932   
236446  Carlyle Carrol Tapsell   M  27.0   182.0     NaN  India  IND  1936   
250689         Max "Maxie" Vaz   M   NaN     NaN     NaN  India  IND  1948   
266934    Sayed Muhammad Yusuf   M   NaN     NaN     NaN  India  IND  1928   

        Season         City   Sport                Event Medal  Gold  Silver  \
4732    Summer    Amsterdam  Hockey  Hockey Men's Hockey  Gold     1       0   
4736    Summer        Tokyo  Hockey  Hockey Men's Hockey  Gold     1       0   
5032    Summer    Amsterdam  Hockey  Hockey Men's Hockey  Gold     1       0   
5033    Summer  Los Angeles  Hockey  Hockey Men's Hockey  Gold     1       0   
5034    Summer       Berlin  Hockey  Hockey Men's Hockey  Gold     1       0   
...        ...          ...     ...                  ...   ...   ...     ...   
233749  Summer       Moskva  Hockey  Hockey Men's Hockey  Gold     1       0   
236445  Summer  Los Angeles  Hockey  Hockey Men's Hockey  Gold     1       0   
236446  Summer       Berlin  Hockey  Hockey Men's Hockey  Gold     1       0   
250689  Summer       London  Hockey  Hockey Men's Hockey  Gold     1       0   
266934  Summer    Amsterdam  Hockey  Hockey Men's Hockey  Gold     1       0   

        Bronze  
4732         0  
4736         0  
5032         0  
5033         0  
5034         0  
...        ...  
233749       0  
236445       0  
236446       0  
250689       0  
266934       0  

[197 rows x 16 columns]

#### :India have got gold medals in Hockey,so its not in the no_gold_medal countries list

### 4.Progressive report of medals by country

In [23]:
crostab=pd.crosstab(index=df.Team,columns=df.Year,values=df.Gold,aggfunc='sum')

In [24]:
crostab.loc['Soviet Union']

Year
1896      NaN
1900      NaN
1904      NaN
1906      NaN
1908      NaN
1912      NaN
1920      NaN
1924      NaN
1928      NaN
1932      NaN
1936      NaN
1948      NaN
1952     36.0
1956     94.0
1960     66.0
1964     88.0
1968     89.0
1972    141.0
1976    148.0
1980    201.0
1984     27.0
1988    168.0
1992      NaN
1994      NaN
1996      NaN
1998      NaN
2000      NaN
2002      NaN
2004      NaN
2006      NaN
2008      NaN
2010      NaN
2012      NaN
2014      NaN
2016      NaN
Name: Soviet Union, dtype: float64

### 5.Tallest athlete

In [25]:
df.nlargest(1,'Height')

Name Sex   Age  Height  Weight   Team  NOC  Year  Season    City  \
265040  Yao Ming   M  20.0   226.0   141.0  China  CHN  2000  Summer  Sydney   

             Sport                        Event Medal  Gold  Silver  Bronze  
265040  Basketball  Basketball Men's Basketball   NaN     0       0       0

#### :Yao Ming from china is the tallest player in the dataset

### 6.Youngest player

In [26]:
df.nsmallest(1,'Age')

Name Sex   Age  Height  Weight  \
142882  Dimitrios Loundras   M  10.0     NaN     NaN   

                                 Team  NOC  Year  Season    City       Sport  \
142882  Ethnikos Gymnastikos Syllogos  GRE  1896  Summer  Athina  Gymnastics   

                                        Event   Medal  Gold  Silver  Bronze  
142882  Gymnastics Men's Parallel Bars, Teams  Bronze     0       0       1

#### :Dimitrios Loundras of age 10 is the youngest player in the dataset

### 7.Countries without female participants

In [27]:
gender_count=pd.crosstab(df.Sex,df.NOC)
gender_count

NOC  AFG  AHO  ALB  ALG  AND  ANG  ANT  ANZ   ARG  ARM  ...  VIE  VIN  VNM  \
Sex                                                     ...                  
F      5   12   27   94   42  129   37    2   643   38  ...   74   11    2   
M    121   67   43  457  127  138   96   84  2654  183  ...   80   31   60   

NOC  WIF  YAR  YEM  YMD   YUG  ZAM  ZIM  
Sex                                      
F      0    0    4    0   378   19  113  
M     20   11   28    5  2205  164  198  

[2 rows x 230 columns]

In [28]:
gender_count01=gender_count.unstack().unstack()

In [29]:
gender_count01

Sex    F     M
NOC           
AFG    5   121
AHO   12    67
ALB   27    43
ALG   94   457
AND   42   127
..   ...   ...
YEM    4    28
YMD    0     5
YUG  378  2205
ZAM   19   164
ZIM  113   198

[230 rows x 2 columns]

In [30]:
countries_without_female_players=gender_count01[gender_count01['F']==0]

In [31]:
countries_without_female_players

Sex  F    M
NOC        
CRT  0   11
NBO  0    2
NFL  0    1
UAR  0  123
UNK  0    2
WIF  0   20
YAR  0   11
YMD  0    5

### 8.Gold medalist in basketball event,1992 summer olympics

In [32]:
df.query("Year==1992 & Sport=='Basketball' & Medal=='Gold' ")

Name Sex   Age  Height  \
14220                          Yelena Viktorovna Baranova   F  20.0   192.0   
14656                                Charles Wade Barkley   M  29.0   198.0   
22150                                      Larry Joe Bird   M  35.0   205.0   
31954                Elen Rafaelovna Bunatyants-Shakirova   F  22.0   190.0   
58646                                Clyde Austin Drexler   M  30.0   200.0   
65566                              Patrick Aloysius Ewing   M  29.0   213.0   
78340                            Irina Yakovlevna Gerlits   F  26.0   190.0   
109625                        Earvin "Magic" Johnson, Jr.   M  32.0   205.0   
110517                             Michael Jeffrey Jordan   M  29.0   198.0   
117541                      Yelena Anatolyevna Khudashova   F  27.0   195.0   
130713                          Christian Donald Laettner   M  22.0   211.0   
147819                                        Karl Malone   M  29.0   205.0   
159784                              Irina Edvinovna Minkh   F  28.0   175.0   
165616                    Christopher Paul "Chris" Mullin   M  28.0   200.0   
189367                             Scottie Maurice Pippen   M  26.0   200.0   
201956                             David Maurice Robinson   M  26.0   216.0   
219191            Yelena Petrovna Shvaybovich (Ksenzhik-)   F  26.0     NaN   
229678                              John Houston Stockton   M  30.0   185.0   
232113         Irina Vladimirovna Sumnikova (Sviridenko-)   F  27.0   170.0   
240647                Maryna Ihorivna Tkachenko (Kopcha-)   F  26.0   185.0   
241949                       Yelena Andonisovna Tornikidu   F  27.0   185.0   
267051  Svetlana Aleksandrova Zaboluyeva-Antipova (-Be...   F  25.0   180.0   
268005                      Nataliya Borisovna Zasulskaya   F  23.0   190.0   
269608            Olena Mykolaivna Zhyrko (-Marenchikova)   F  24.0   180.0   

        Weight           Team  NOC  Year  Season       City       Sport  \
14220     80.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
14656    114.0  United States  USA  1992  Summer  Barcelona  Basketball   
22150    100.0  United States  USA  1992  Summer  Barcelona  Basketball   
31954     86.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
58646    101.0  United States  USA  1992  Summer  Barcelona  Basketball   
65566    109.0  United States  USA  1992  Summer  Barcelona  Basketball   
78340     72.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
109625   100.0  United States  USA  1992  Summer  Barcelona  Basketball   
110517    90.0  United States  USA  1992  Summer  Barcelona  Basketball   
117541    78.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
130713   107.0  United States  USA  1992  Summer  Barcelona  Basketball   
147819   116.0  United States  USA  1992  Summer  Barcelona  Basketball   
159784    64.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
165616    98.0  United States  USA  1992  Summer  Barcelona  Basketball   
189367   102.0  United States  USA  1992  Summer  Barcelona  Basketball   
201956   107.0  United States  USA  1992  Summer  Barcelona  Basketball   
219191     NaN   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
229678    79.0  United States  USA  1992  Summer  Barcelona  Basketball   
232113    60.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
240647     NaN   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
241949    73.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
267051    70.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
268005    95.0   Unified Team  EUN  1992  Summer  Barcelona  Basketball   
269608     NaN   Unified Team  EUN  1992  Summer  Barcelona  Basketball   

                                Event Medal  Gold  Silver  Bronze  
14220   Basketball Women's Basketball  Gold     1       0       0  
14656     Basketball Men's Basketball  Gold     1       0       0

#### :male category-USA ,female -EUN

### 9.Age category wise medal count

In [33]:
age_wise_medal_count=df.groupby(['Medal'])['Age'].value_counts()

In [34]:
age_wise_medal_count.unstack().unstack().unstack()

Medal  Bronze  Gold  Silver
Age                        
10.0      1.0   NaN     NaN
11.0      NaN   NaN     1.0
12.0      3.0   NaN     3.0
13.0      2.0   7.0     7.0
14.0     18.0  27.0    30.0
...       ...   ...     ...
68.0      NaN   NaN     2.0
69.0      1.0   NaN     1.0
71.0      1.0   NaN     1.0
72.0      1.0   NaN     1.0
73.0      NaN   NaN     1.0

[61 rows x 3 columns]

### 10.Oldest person who won gold medal

In [35]:
Gold=df.query("Medal=='Gold'")
Gold_oldest=Gold[Gold['Age']==Gold.Age.max()]
Gold_oldest

Name Sex   Age  Height  Weight           Team  NOC  Year  \
105199    Charles Jacobus   M  64.0     NaN     NaN  United States  USA  1904   
233390  Oscar Gomer Swahn   M  64.0     NaN     NaN         Sweden  SWE  1912   

        Season       City     Sport  \
105199  Summer  St. Louis     Roque   
233390  Summer  Stockholm  Shooting   

                                                   Event Medal  Gold  Silver  \
105199                               Roque Men's Singles  Gold     1       0   
233390  Shooting Men's Running Target, Single Shot, Team  Gold     1       0   

        Bronze  
105199       0  
233390       0